dbms/src/Functions/greatCircleArc.cpp

```diff
@@ -0,0 +1,74 @@
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+#include <ext/range.h>
+#include <cmath>
+
+namespace DB {
+
+class FunctionGreatCircleArc : public IFunction {
```

Comment:
We already have this function, it is named `greatCircleDistance`.

Reply:
It looks the same up to coefficient.

Reply:
No, it isn't. Both name and documentation of `greatCircleDistance` confuses. At fact `greatCircleDistance` returns metric distance on the surface of ellipsoid, not sphere, it can be proven by source code or simple SQL query (current master):
```sql
SELECT greatCircleDistance(0., 0., 0., 1.) / greatCircleDistance(0., 89., 0., 90.) AS x
```
```text
┌──────────────────x─┐
│ 0.9898663431043628 │
└────────────────────┘
```
Probably `greatCircleDistance` should be renamed, but is not related to this PR.

Reply:
The aim of `greatCircleArc` is an arc distance on sphere

Reply:
Probably the behaviour of `greatCircleDistance` has been changed in ae2f5ad from "great circle distance" to "ellipsoid distance". I'm not sure but the code in the commit looks like realisation of [WGS 84](https://en.wikipedia.org/wiki/World_Geodetic_System#WGS84)

Reply:
No, it's still for sphere.
The calculation on ellipsoid would be way more difficult.

Reply:
@stavrolia is responsible for this function.

Reply:
I opened Issue #8080 to discuss it

Reply:
If it would be for sphere how can it give 1% difference for equator and pole regions? You can compare output of `greatCircleDistance` with some WGS calculator and with spherical formula. Results are much closer to ellipsoid than to sphere. Let me compare with https://www.cqsrg.org/tools/GCDistance/ WGS web calculator

**Current realisation**
```sql
SELECT greatCircleDistance(0., 0., 0., 1.)
```
```text
┌─greatCircleDistance(0., 0., 0., 1.)─┐
│                           110567.33 │
└─────────────────────────────────────┘
```
**Old realisation**
```sql
SELECT greatCircleDistance(0., 0., 0., 1.)
```
```text
┌─greatCircleDistance(0., 0., 0., 1.)─┐
│                  111226.30000000089 │
└─────────────────────────────────────┘
```
**SQSRG**: 110574.389

**Current realisation**
```sql
SELECT greatCircleDistance(0., 89., 0., 90.)
```
```text
┌─greatCircleDistance(0., 89., 0., 90.)─┐
│                             111699.25 │
└───────────────────────────────────────┘
```
**Old realisation**
```sql
SELECT greatCircleDistance(0., 89., 0., 90.)
```
```text
┌─greatCircleDistance(0., 89., 0., 90.)─┐
│                    111226.30000000105 │
└───────────────────────────────────────┘
```
**SQSRG**: 111693.865

Reply:
Comment to the new `greatCircleDistance` says that the origin of this code is 
https://github.com/sphinxsearch/sphinx/blob/409f2c2b5b2ff70b04e38f92b6b1a890326bad65/src/sphinxexpr.cpp#L3825.

Tests of this project compare the output of this code with [Vincenty's formula](https://en.wikipedia.org/wiki/Vincenty%27s_formulae) which is approximation for ellipsoid distance, WGS 84 ellipsoid is used:
https://github.com/sphinxsearch/sphinx/blob/409f2c2b5b2ff70b04e38f92b6b1a890326bad65/src/tests.cpp#L3577
https://github.com/sphinxsearch/sphinx/blob/409f2c2b5b2ff70b04e38f92b6b1a890326bad65/src/tests.cpp#L3682

Reply:
Also see slides which are mentioned in comment to `greatCircleDistance`:
https://github.com/ClickHouse/ClickHouse/files/3324740/1_._._GEODIST_._.pdf

Slides say that the origin Sphinx code is written for geoid distance

Reply:
@alexey-milovidov in the light of #8086, I propose to remove `*Arc` functions from the PR and add functions to convert between degrees and metric distance on spherical Earth. What do you think of that?

Reply:
I've removed `greatCircleArc`